### PR TITLE
POLIO-952:  BACKEND - Preparedness tab: remove "Surge data" part

### PR DESCRIPTION
**Pull request by hakifran**
Explain what problem this PR is resolving
- Surge datas(fields) are not needed anymore
Related JIRA tickets : [POLIO-952](https://bluesquare.atlassian.net/browse/POLIO-952)

## Self proofreading checklist

- [x] Did I use eslint and black formatters
- [x] Is my code clear enough and well documented
- [ ] Are my typescript files well typed
- [ ] New translations have been added or updated if new strings have been introduced in the frontend
- [x] My migrations file are included
- [ ] Are there enough tests
- [ ] Documentation has been included (for new feature)

## Changes
- Remove the surge model
- Remove two campaign fields: surge_spreadsheet_url and country_name_in_surge_spreadsheet

[POLIO-952]: https://bluesquare.atlassian.net/browse/POLIO-952?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by hakifran** [comment]
@cypress

**Comment by hakifran** [comment]
@cypress

**Comment by hakifran** [comment]
@cypress

File: plugins/polio/js/src/constants/translations/fr.json
Comment (codeKameleon) [inline plugins/polio/js/src/constants/translations/fr.json]: Shouldn't we keep the translations ? As they will still be used in the campaign history for older campaigns

https://user-images.githubusercontent.com/25134301/233022205-df265c97-bd6a-41a4-af31-3bce8e7d202a.mov

**Review by codeKameleon — CHANGES_REQUESTED** [review]
Except for the translations (see comment) it seems good to me 👍

File: plugins/polio/js/src/forms/PreparednessForm.js
Comment (beygorghor) [inline plugins/polio/js/src/forms/PreparednessForm.js]: moment is not used anymore

File: plugins/polio/js/src/forms/PreparednessForm.js
Comment (beygorghor) [inline plugins/polio/js/src/forms/PreparednessForm.js]: Not used anymore

File: plugins/polio/js/src/forms/PreparednessForm.js
Comment (beygorghor) [inline plugins/polio/js/src/forms/PreparednessForm.js]: Not used anymore

Comment (beygorghor) [reply plugins/polio/js/src/forms/PreparednessForm.js]: please check you eslint for those lines

File: plugins/polio/js/src/forms/PreparednessForm.js
Comment (beygorghor) [inline plugins/polio/js/src/forms/PreparednessForm.js]: eslint issue here

**Review by beygorghor — CHANGES_REQUESTED** [review]
Nice, just remove unused imports and check Eslint please

File: plugins/polio/js/src/constants/translations/fr.json
Comment (hakifran) [reply plugins/polio/js/src/constants/translations/fr.json]: @codeKameleon As I removed all field in the DB, these translation won't be needed anymore even for the older campaigns

**Comment by hakifran** [comment]
@cypress

**Review by beygorghor — APPROVED** [review]
Thx !

**Review by codeKameleon — APPROVED** [review]

File: plugins/polio/models.py
Comment (quang-le) [inline plugins/polio/models.py:440]: To double-check with @madewulf or @mathvdh but it seems to me @olethanh 's policy with this model is to deprecate fields but not delete them

File: plugins/polio/models.py
Comment (quang-le) [inline plugins/polio/models.py:733]: cf other comment

**Review by quang-le — CHANGES_REQUESTED** [review]

File: plugins/polio/models.py
Comment (madewulf) [reply plugins/polio/models.py:440]: In this case, I think we can really delete, but we have to include the corresponding migration

Comment (hakifran) [reply plugins/polio/models.py:440]: @quang-le , @madewulf there are three migrations which have been created to remove the fields: 
- plugins/polio/migrations/0128_remove_campaign_surge_spreadsheet_url.py
- plugins/polio/migrations/0129_remove_campaign_country_name_in_surge_spreadsheet.py
- plugins/polio/migrations/0130_delete_surge.py

File: plugins/polio/js/src/components/CreateEditDialog.js
Comment (quang-le) [inline plugins/polio/js/src/components/CreateEditDialog.js]: There's an unused import to delete related to this

**Review by quang-le — APPROVED** [review]

File: plugins/polio/js/src/components/CreateEditDialog.js
Comment (hakifran) [reply plugins/polio/js/src/components/CreateEditDialog.js]: @quang-le I've removed it. thank you

File: plugins/polio/js/src/constants/translations/fr.json
Comment (olethanh) [reply plugins/polio/js/src/constants/translations/fr.json]: Theses field would still be in the saved json history dump

Comment (olethanh) [reply plugins/polio/js/src/constants/translations/fr.json]: but from the little they have been used we could probably remove them. 

@codeKameleon Maybe it would make send to keep an dict or an array with all the string used in the history page (old and current field) ? So they don't get removed by mistakes

**Comment by olethanh** [comment]
Test locally and works for me. Asked the PowerBi team (Rita) to check if the removal won't cause problem for their api, **we need to wait for their confirmation BEFORE merging** 

Also suggest  that during release we deploy on polio staging before hand to check everything works, since this is quite the DB migration

**Comment by olethanh** [comment]
After discussion, since we can break the API for powerbi in the short term,  let split this in a separate frontend and backend PR.

**Comment by hakifran** [comment]
@cypress

**Comment by hakifran** [comment]
@olethanh this PR is in draft and it contains only backend changes related to remove "Surge data"

**Comment by hakifran** [comment]
@cypress

**Comment by hakifran** [comment]
@cypress

**Review by madewulf — APPROVED** [review]

**Pull request merged**
